cloudferrylib/os/actions/check_networks.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+import ipaddr
+import collections
+import exceptions
+from cloudferrylib.base.action import action
+from cloudferrylib.utils import utils
+
+
+LOG = utils.get_log(__name__)
+
+
+class CheckNetworks(action.Action):
+    def run(self, **kwargs):
+        LOG.debug("Start 'check_netwroks' action")
+        src_net = self.src_cloud.resources[utils.NETWORK_RESOURCE]
+        dst_net = self.dst_cloud.resources[utils.NETWORK_RESOURCE]
+        search_opts = kwargs.get('search_opts_tenant', {})
+        src_info = NetworkInfo(src_net.read_info(**search_opts))
+        dst_info = NetworkInfo(dst_net.read_info(**search_opts))
+        for network in src_info.get_networks():
```

Comment:
Please consider refactoring to make the process more linear, it has too many levels of indentation right now (I counted 7).